### PR TITLE
chore(helpers): move unique store into faker instance

### PR DESCRIPTION
**Pull request by Shinigami92**
references #1791

### CODE REVIEW

**Review by ST-DDT — COMMENTED** [review]
Should we merge this only if it helps with the tree shake-ability or anyway?

**Comment by Shinigami92** [comment]
> Should we merge this only if it helps with the tree shake-ability or anyway?

I'm also not so sure

**Comment by xDivisionByZerox** [comment]
Are these tests still relevant in their current setup (do they still have to be separated)?

https://github.com/faker-js/faker/blob/3fc7bf1b24ed67696b4824abcd9fa14af43cb66d/test/helpers.spec.ts#L1201-L1203

File: src/modules/helpers/index.ts [hunk 1]
```diff
@@ -83,6 +83,22 @@ function getRepetitionsBasedOnQuantifierParameters(
...
+   */
+  private uniqueStore: Record<RecordKey, RecordKey> = {};
+
+  /**
+   * Global exclude list of results.
+   * Defaults to nothing excluded.
+   *
+   * @internal
+   */
+  private uniqueExclude: RecordKey[] = [];
```
Comment (ST-DDT) [inline src/modules/helpers/index.ts]: AFAICT this can be inlined because it is readonly.

**Comment by Shinigami92** [comment]
> Are these tests still relevant in their current setup (do they still have to be separated)?
> 
> https://github.com/faker-js/faker/blob/3fc7bf1b24ed67696b4824abcd9fa14af43cb66d/test/helpers.spec.ts#L1201-L1203

I see what you mean, but let's ignore it for now as the method is deprecated anyway
The tests are just executed once instead of multiple times, but the globals are still bound internally to the faker instance

**Review by xDivisionByZerox — APPROVED** [review]
> I see what you mean, but let's ignore it for now as the method is deprecated anyway
The tests are just executed once instead of multiple times, but the globals are still bound internally to the faker instance

Yes, but you could instantiate a new Faker object in every test which wasn't possible before.

I don't really like to ignore it, but the fact that it is deprecated seems reasonable.

**Review by ST-DDT — APPROVED** [review]
The jsdocs aren't ideal, but I don't know how to phrase them any better.

**Pull request merged**